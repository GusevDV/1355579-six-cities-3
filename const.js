export const OfferType = {
  APARTMENT: `apartment`,
  ROOM: `room`,
  HOUSE: `house`,
  HOTEL: `hotel`,
};
export const MapSetting = {
  ZOOM: 12,
  ICON_URL: `/img/pin.svg`,
  ICON_URL_ACTIVE: `/img/pin-active.svg`,
  ICON_SIZE: [30, 30]
};

export const ErrorMessage = {
  NETWROK_ERROR: `–ù–∞–º –Ω–µ —É–¥–∞–ª–æ—Å—å –ø–æ–ª—É—á–∏—Ç—å –¥–∞–Ω–Ω—ã–µ —Å —Å–µ—Ä–≤–µ—Ä–∞ üò¢`
};

export const mapDisplayType = {
  CITIES: `cities`,
  PROPERTY: `propery`,
};

export const monthNames = [`January`, `February`, `March`, `April`, `May`, `June`, `July`, `August`, `September`, `October`, `November`, `December`];

export const SortIndex = {
  PRICE_LOW_TO_HIGH: 1,
  PRICE_HIGH_TO_LOW: 2,
  TOP_RATED_FIRST: 3
};
export const sortTypes = [`Popular`, `Price: low to high`, `Price: high to low`, `Top rated first`];

export const MAX_CITIES_COUNT = 6;
export const MAX_REVIEWS_COUNT = 10;
export const BASE_API_URL = `https://htmlacademy-react-3.appspot.com/six-cities`;
