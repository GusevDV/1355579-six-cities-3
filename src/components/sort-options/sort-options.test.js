import React from "react";
import renderer from "react-test-renderer";
import SortOptions from "./sort-options.jsx";
import {sortTypes as types} from "../../const.js";

it(`Should active SortOptions component render correctly`, () => {

  const render = renderer.create(
      <SortOptions
        sortTypes={types}
        isActive={true}
        activeItem={1}
        onItemClick={()=>{}}
        onToggle={()=>{}}
      />
  )
  .toJSON();

  expect(render).toMatchSnapshot();
});

it(`Should non active SortOptions component render correctly`, () => {

  const render = renderer.create(
      <SortOptions
        sortTypes={types}
        isActive={false}
        activeItem={0}
        onItemClick={()=>{}}
        onToggle={()=>{}}
      />
  )
  .toJSON();

  expect(render).toMatchSnapshot();
});
