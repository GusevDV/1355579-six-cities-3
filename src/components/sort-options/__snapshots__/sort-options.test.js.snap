// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should active SortOptions component render correctly 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
  onClick={[Function]}
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    tabIndex="0"
  >
    Price: low to high
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom places__options--opened"
  >
    <li
      className="places__option "
      onClick={[Function]}
      tabIndex={0}
    >
      Popular
    </li>
    <li
      className="places__option places__option--active"
      onClick={[Function]}
      tabIndex={1}
    >
      Price: low to high
    </li>
    <li
      className="places__option "
      onClick={[Function]}
      tabIndex={2}
    >
      Price: high to low
    </li>
    <li
      className="places__option "
      onClick={[Function]}
      tabIndex={3}
    >
      Top rated first
    </li>
  </ul>
</form>
`;

exports[`Should non active SortOptions component render correctly 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
  onClick={[Function]}
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    tabIndex="0"
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom "
  >
    <li
      className="places__option places__option--active"
      onClick={[Function]}
      tabIndex={0}
    >
      Popular
    </li>
    <li
      className="places__option "
      onClick={[Function]}
      tabIndex={1}
    >
      Price: low to high
    </li>
    <li
      className="places__option "
      onClick={[Function]}
      tabIndex={2}
    >
      Price: high to low
    </li>
    <li
      className="places__option "
      onClick={[Function]}
      tabIndex={3}
    >
      Top rated first
    </li>
  </ul>
</form>
`;
